### PR TITLE
docs: add documentation for realm migration process

### LINKED ISSUE #1267: Document realm migration steps
### Description
Document steps required for realm migration. 

### Motivation
To make it easier for all developers to do migration whenever it's required.

### Requirements
Create a markdown document enlisting all steps required for performing migration.

### Am I planning to do it myself with a PR?
Yes.

### PR DESCRIPTION
**Pull request by laumair**
# Description

Fixes https://github.com/iotaledger/trinity-wallet/issues/1267

## Type of change

- Documentation

# How Has This Been Tested?

N/A

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [ ] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: docs/dev/realm.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+2. Create `index.js` inside the newly created directory in `shared/schemas/`.
+
+3. Export the updated schemas as default. For example, if the updated schema has added a new property to `WalletSettings` schema, the default export should look like:
+
+```javascript
+import merge from 'lodash/merge';
+import v6Schema from '../v6';
+
+export default v4Schema.map((schema) => {
```
Comment (rajivshah3) [inline docs/dev/realm.md]:
```suggestion
export default v6Schema.map((schema) => {
```

**Comment by rajivshah3** [comment]
The failing Snyk test can be resolved by merging the latest commits from `develop`

File: docs/dev/realm.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+**Note: Ensure to always extend the most recent schema (as shown in the above example). If the next schema version is going to be 7, we should make sure that we extend the most recent i.e., v6 schema.**
+
+4. Export a realm [migration](https://realm.io/docs/javascript/latest/#migrations) function. 
+
+5. Import updated schema and migration function in `shared/schema/index.js`. For example, if the updated schema is `v7`, the import in `shared/schema/index.js` should look like:
+
+```javascript
+import v7Schema, { migration as v5Migration } from './v5';
```
Comment (rajivshah3) [inline docs/dev/realm.md]: Whoops, I think I missed this when I reviewed earlier

```suggestion
import v7Schema, { migration as v7Migration } from './v7';
```

**Comment by laumair** [comment]
@rajivshah3 https://github.com/iotaledger/trinity-wallet/pull/1999/commits/3dd51ac4fecbb134fc78cadb59c3e24b24b096f8

**Review by rajivshah3 — APPROVED** [review]

**Pull request merged**